### ISSUE TITLE
locationTemplate functionality appears to depend on project's AndroidManifest.xml's android:name

### ISSUE THREAD

**Issue by gerryhocks** [transistorsoft/flutter_background_geolocation#659]
## Your Environment
* Plugin version: 4.3.0 and git:headless-notificationaction
* Platform: Android (and iOS as an aside)
* OS version: Android 10, iOS 15.0.1
* Device manufacturer / model: ulefone/notep 9, Apple/iPhone6S
* Flutter info (`flutter doctor`): 
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel dev, 2.6.0-11.0.pre, on Microsoft Windows [Version 10.0.19043.1288], locale en-GB)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Android Studio (version 2020.3)
[√] Connected device (3 available)
• No issues found!
* Plugin config:      

 
```dart <-- Syntax highlighting: DO NOT REMOVE -->
locationTemplate: '{"lat":<%= latitude %>,"lon":<%= longitude %>,"heading":<%= heading %>,"speed":<%= speed %>,"timestamp":"<%= timestamp %>"}',
 geofenceTemplate: '{"lat":<%= latitude %>,"lon":<%= longitude %>,"heading":<%= heading %>,"speed":<%= speed %>,"timestamp":"<%= timestamp %>"}',
```

## Expected Behavior

The locationTemplate functionality should not depend on the AndroidManifest's android:name property.
## Actual Behavior

The locationTemplate functionality appears to depend on the AndroidManifest's android:name property.
## Steps to Reproduce

1. Change the locationTemplate and run the example app
2. Note the posted JSON structure
3. Change the AndroidManifest as described and rerun the example app
4. Note the posted JSON structure

## Context

I have been modifying the existing example application since it closely matches my requirements.

However, on Android I could not get locationTemplate to change the posted JSON. I reviewed the issues here and determined it was not due to stale or unset sqlite entries.

After experimentation, I found that changing the example's AndroidManifest.xml android:name setting from the provided ".Application" to  "io.flutter.app.FlutterApplication" cause the locationTemplate to start working and posting templated JSON.

Subsequently changing it back to ".Application" (with no other code changes) causes the full JSON to be posted again.

In summary, by changing this line - and only this line - the application sends different JSON to the target service. In all other respects, the application works identically.

I understand the examples are to be rebuilt, but this might assist others basing a new application on the example application.

As an aside, I cannot get the sample application to use the locationTemplate on iOS, but need to do some more investigation before raising an issue regarding this.
## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by christocracy** [comment]
> locationTemplate functionality appears to depend on project's AndroidManifest.xml's android:name

False

**Comment by gerryhocks** [comment]
Yep. I understand all of that. I'm well down the path of modifying the examples app. 

I've tried reset:true, reset(), ready() and setConfig(). I've wiped the app several times.

I have a situation now where the JSON changes purely when changing that one line in the manifest, with no code changes.

**Comment by gerryhocks** [comment]
I see. It's false. Observable, repeatable, but false. Okies. Anyhow, I did come up with a solution (a false one) and I'm happy with it, even though it's false.

I'll figure out the iOS issue but I am not sure I'll post an issue about it. You are an odd fellow.
